pkg/mapper/dynamicfile/dynamicfile.go

```diff
@@ -4,22 +4,31 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"os"
-	"strings"
-	"sync"
-	"time"
-
 	"github.com/fsnotify/fsnotify"
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"os"
 	"sigs.k8s.io/aws-iam-authenticator/pkg/arn"
 	"sigs.k8s.io/aws-iam-authenticator/pkg/config"
+	"strings"
+	"sync"
+	"time"
 )
 
+type RoleMappingV2 struct {
```

Comment:
I would expect adding a field to be backwards compatible.  Are there any situations that we need a new struct for?

Reply:
Yep I can add the new field into existing RoleMapping struct. But existing mode user might wonder what this field is and what value shall they put in when they create new entries for existing mode. Hence I introduced a new struct for dynamic file and minimize the impact.